### PR TITLE
[NIL] Update NIL Indicator migration fields

### PR DESCRIPTION
**Pull request by mchalejeiwc**
Primarily this is just to do the following:
- remove published date field
- replace 'descriptor' field with four new fields
(this is already reflected in the main project)

There are a couple of 'robustness' fixes. The poi package likes to return null for empty cells, so I'm now handling this, and we also now escape a few special characters for the JSON output (in the scope of this project didn't feel that pulling in a new TP library was necessary). Additionally some of the provided input sheets are appended by blank rows, so simply break out.

### CODE REVIEW

File: src/main/java/uk/nhs/digital/ps/migrator/model/hippo/NationalIndicatorMigrator.java [hunk 1]
```diff
@@ -5,8 +5,10 @@
 import java.util.Iterator;
 import java.util.List;
 
-import org.apache.poi.ss.usermodel.Row;
+import javax.sql.DataSource;
```
Comment (ziemsky) [inline src/main/java/uk/nhs/digital/ps/migrator/model/hippo/NationalIndicatorMigrator.java]: Seems that this has been imported by accident.

File: src/main/java/uk/nhs/digital/ps/migrator/model/hippo/NationalIndicatorMigrator.java [hunk 2]
```diff
@@ -5,8 +5,10 @@
...
 
+import org.apache.poi.ss.usermodel.Row;
+import org.slf4j.LoggerFactory;
```
Comment (ziemsky) [inline src/main/java/uk/nhs/digital/ps/migrator/model/hippo/NationalIndicatorMigrator.java]: While removing the DataSource import, would you mind cutting this one out too, Jon?

File: src/main/java/uk/nhs/digital/ps/migrator/model/hippo/XlsxReader.java [hunk 3]
```diff
@@ -91,18 +94,35 @@ public Integer getColumnIndex(String columnName){
...
+
+        String rawValue = cell.getStringCellValue();
+        if (rawValue == null)
+            return null;
+
+
+        return TextHelper.escapeSpecialCharsForJson(rawValue);
     }
 
     /**l
```
Comment (ziemsky) [inline src/main/java/uk/nhs/digital/ps/migrator/model/hippo/XlsxReader.java]: I think this redundant 'l' is a result of locking the machine via WinKey+L - happens to me all the time :smile: (I know it's from another commit, just spotted it now).

**Review by ziemsky — APPROVED** [review]
Looking good with just a few suggestions for cosmetic changes (just to pick on something to show I'm paying attention :stuck_out_tongue_closed_eyes: ).

The suggestions are totally optional - I'm approving the PR, leaving it to you to decide if you want to follow my comments or not :wink:

File: src/main/java/uk/nhs/digital/ps/migrator/model/hippo/XlsxReader.java (hunk 3 above)
Comment (mchalejeiwc) [reply src/main/java/uk/nhs/digital/ps/migrator/model/hippo/XlsxReader.java]: From now on I shall call you Hawkeye.

**Comment by mchalejeiwc** [comment]
I've actioned your cosmetic suggestions :)

**Pull request merged**